cnd/src/swap_protocols/swap_id.rs

```diff
@@ -38,3 +38,39 @@ impl fmt::Display for LocalSwapId {
         self.0.fmt(f)
     }
 }
+
+/// This is a swap identifier created by Bob and shared with Alice via the
+/// network communication protocols.
+#[derive(Serialize, Deserialize, Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct SharedSwapId(Uuid);
+
+impl Default for SharedSwapId {
+    fn default() -> Self {
+        SharedSwapId(Uuid::new_v4())
+    }
+}
+
+impl FromStr for SharedSwapId {
+    type Err = uuid::Error;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Uuid::from_str(s).map(SharedSwapId)
+    }
+}
+
+impl From<Uuid> for SharedSwapId {
```

Comment:
I don't think this trait is needed. `Uuid` should just be an internal implementation detail.

Reply:
Removed, thanks man.